src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionsRefundQueueConsumer.kt

```diff
@@ -27,6 +31,53 @@ import org.springframework.stereotype.Service
 import reactor.core.publisher.Mono
 import reactor.kotlin.core.publisher.switchIfEmpty
 
+private class RefundQueueEvent(
+  private val refundRequestedEvent: QueueEvent<TransactionRefundRequestedEvent>?,
+  private val refundRetriedEvent: QueueEvent<TransactionRefundRetriedEvent>?,
+  private val expiredEvent: QueueEvent<TransactionExpiredEvent>?
+) {
+  companion object {
+    fun fromRefundRequested(
+      refundRequestedEvent: QueueEvent<TransactionRefundRequestedEvent>
+    ): RefundQueueEvent = RefundQueueEvent(refundRequestedEvent, null, null)
+
+    fun fromRefundRetried(
+      refundRetriedEvent: QueueEvent<TransactionRefundRetriedEvent>
+    ): RefundQueueEvent = RefundQueueEvent(null, refundRetriedEvent, null)
+
+    fun fromExpired(expiredEvent: QueueEvent<TransactionExpiredEvent>): RefundQueueEvent =
+      RefundQueueEvent(null, null, expiredEvent)
+  }
+
+  init {
+    /*
+     * Require that only one variant is not null
+     */
+    val conditions =
+      listOf(refundRequestedEvent != null, refundRetriedEvent != null, expiredEvent != null)
+
+    require(conditions.filter { it }.size == 1) {
```

Comment:
Add test to cover that IllegalArgumentException is effectively thrown for this class instantiated with 2 events valued just to make this logic modification covered by a test